### PR TITLE
Fix Modrinth query when Quilt is in use

### LINKED ISSUE #446: Cant use Modrinth with Quilt
### Operating System

_No response_

### Description of bug

Mod downloader will not show anything under the modrinth section when Quilt is set as the modloader

### Steps to reproduce

create an instance with quilt go to edit instance > mods > download mods 

Modrinth will be blank like this:
unknown-8

Curseforge works as expected

### Suspected cause

No quilt filter option and not fallbacking to fabric

### This issue is unique

- [X] I have searched the issue tracker and did not find an issue describing my bug.

### PR DESCRIPTION
**Pull request by Scrumplex**
Closes #446

### CODE REVIEW

**Comment by Scrumplex** [comment]
CC: @flowln 
Do you think this is fine? It's a lot more code now, but it should be very flexible now.

**Comment by DioEgizio** [comment]
I think this changes are also necessary in the case of a mod filter based on the loader

**Review by ZekeZDev — APPROVED** [review]

**Review by DioEgizio — APPROVED** [review]
this seems to work

File: launcher/ui/pages/modplatform/modrinth/ModrinthPage.cpp [hunk 1]
```diff
@@ -60,9 +61,19 @@ ModrinthPage::ModrinthPage(ModDownloadDialog* dialog, BaseInstance* instance)
...
 }
 
-auto ModrinthPage::validateVersion(ModPlatform::IndexedVersion& ver, QString mineVer, QString loaderVer) const -> bool
+auto ModrinthPage::validateVersion(ModPlatform::IndexedVersion& ver, QString mineVer, ModAPI::ModLoaderType loader) const -> bool
 {
-    return ver.mcVersion.contains(mineVer) && ver.loaders.contains(loaderVer);
+    auto loaderStrings = ModrinthAPI::getModLoaderStrings(loader);
+
+    auto loaderCompatible = false;
+    for (auto loader : ver.loaders)
```
Comment (flowln) [inline launcher/ui/pages/modplatform/modrinth/ModrinthPage.cpp]: Can you change this `loader` to something else? It shadows (is this the right term?) the function parameter, and makes it harder to see which one is being referenced later on imo

File: launcher/ui/pages/modplatform/flame/FlameModPage.cpp [hunk 2]
```diff
@@ -38,6 +38,7 @@
 
 #include "FlameModModel.h"
 #include "ui/dialogs/ModDownloadDialog.h"
+#include <qglobal.h>
```
Comment (flowln) [inline launcher/ui/pages/modplatform/flame/FlameModPage.cpp]: This import is not needed (nvim moment? :^) )

**Review by flowln — COMMENTED** [review]
looks fine to me, just some minor thingies :)

File: launcher/ui/pages/modplatform/flame/FlameModPage.cpp (hunk 2 above)
Comment (Scrumplex) [reply launcher/ui/pages/modplatform/flame/FlameModPage.cpp]: Yes nvim moment

**Review by txtsd — APPROVED** [review]

**Review by flowln — APPROVED** [review]

**Review by d-513 — APPROVED** [review]

**Pull request merged**